src/main/resources/db/migration/postgres/V2.4.1.0__data_column_order_states.sql

```diff
@@ -0,0 +1,338 @@
+-- As part of ACAS-413
+-- Setup a temp table with all the existing experiment columns
+-- We create an artificial "column_order" that will only be used for experiments
+-- that are missing the 'data column order' states
+create table tmp_expt_columns as
+    select *, row_number() over (partition by experiment_id order by column_name) as column_order
+    from (select distinct experiment_id,
+                    ls_type as column_type,
+                    ls_kind as column_name,
+                    unit_kind as units,
+					tested_conc as concentration,
+                    tested_conc_unit as conc_units,
+					not public_data as hidden
+	-- we use the p_api_analysis_group_results view because it includes hidden data
+    from p_api_analysis_group_results paagr) a;
+
+--Create an ls_transaction to tie all our states and values to
+insert into ls_transaction (id, comments, recorded_date, version, recorded_by)
+	select nextval('ls_transaction_pkseq'), 'V2.4.1.0 data column order states migration', now(), 0, 'ACAS';
+
+-- Part 1: Fill in 'data column' DDict Values based on saved data
+-- Create a reusable function to insert missing ddict values
+create or replace function add_data_column_order_ddict_value(val varchar, ls_kind varchar) returns bigint as $$
+	insert into ddict_value (id, ignored, code_name, label_text, ls_type, ls_kind, ls_type_and_kind, short_name, version)
+		select
+			nextval('ddict_value_pkseq') as id,
+			false as ignored,
+			'DDICT-' || lpad(nextval('labelseq_1_ddict_id_codename_document_datadictionary'):: text, 6, '0'::text) as code_name,
+			val as label_text,
+			'data column' as ls_type,
+			ls_kind as ls_kind,
+			'data column_'||ls_kind as ls_type_and_kind,
+			val as short_name,
+			0 as version
+		where not exists (
+			select * from ddict_value dv 
+				where dv.ls_type = 'data column' 
+					and dv.ls_kind = ls_kind 
+					and dv.ignored = false 
+					and dv.short_name = val
+			)
+		returning id
+	$$ language sql;
+
+--column name
+select add_data_column_order_ddict_value(column_name, 'column name'::varchar) from
+(select distinct column_name from tmp_expt_columns) a;
+--units
+select add_data_column_order_ddict_value(units, 'column units'::varchar) from
+(select distinct units from tmp_expt_columns) a where a.units is not null;
+--conc units
+select add_data_column_order_ddict_value(conc_units, 'column conc units'::varchar) from
+(select distinct conc_units from tmp_expt_columns) a where a.conc_units is not null;
```

Comment:
Is there a reason only column name, units, and conc units are being filled in here?

Reply:
Are there other data dictionaries that we need to fill in?
"column type" is a fixed set of options that'll be filled in by the ConfJSON. Concentration and Time are numeric (no data dictionary). I did skip "time units" because I don't think it's really being used.

Reply:
Got it, that makes sense! Time units was the only one that I could think of.